### DISCUSSION TITLE
Best Practices for Secure & Maintainable Raw SQL in Next.js (Avoiding ORMs)

### DISCUSSION THREAD

**Discussion by llaalgol** [vercel/next.js#56213] [Help] [answered]
### Summary

Seeking tips for secure, high-performance SQL in Next.js. Avoiding ORMs for control. Any experiences or library recommendations?

**Comment by leerob** [comment, marked as answer]
One example you can explore is Vercel Postgres: https://vercel.com/docs/storage/vercel-postgres

It allows you to safely use string templates: https://vercel.com/docs/storage/vercel-postgres/sdk#preventing-sql-injections
